source/MRMesh/MRICP.cpp

```diff
@@ -381,29 +381,36 @@ AffineXf3f ICP::calculateTransformation()
     float minDist = std::numeric_limits<float>::max();
     int badIterCount = 0;
     resultType_ = ICPExitType::MaxIterations;
+    AffineXf3f bestIterXf = flt_.xf;
```

Comment:
better `bestXf`, or `minXf` considering that we have `minDist`
Or simply avoid this variable and write directly in `flt_.xf` as you do in MultiwayICP